documentation/RLC-quickstart.md

```diff
@@ -16,96 +14,100 @@ If you are the first time to prepare the SDK, please follow the Azure SDK guidan
    The generated libraries should be in the [azure-sdk-for-js](https://github.com/Azure/azure-sdk-for-js) repo, so fork and clone it in your local then the absolute path is called **${SDK_REPO_ROOT} folder**.
 
 1. Project Folder Structure.  
-   Normally, the folder structure would be something like `sdk/{servicename}/{servicename}-{modulename}-rest`. For example, we have `sdk/agrifood/agrifood-farming-rest` folder for Farmbeats account modules under {SDK_REPO_ROOT}. That folder will be your **${PROJECT_ROOT} folder**.  
+   Normally, the folder structure would be something like `sdk/{servicename}/{servicename}-{modulename}-rest`. For example, we have `sdk/agrifood/agrifood-farming-rest` folder for Farmbeats account modules under {SDK_REPO_ROOT}. That folder will be your **${PROJECT_ROOT} folder**.
 1. Package Name Convention.  
    The package name for RLC is something like `@azure-rest/{servicename}-{modulename}`. For example, the package name for Farmbeats module is `@azure-rest/agrifood-farming`.
 
-
 # How to generate RLC
 
 We are working on to automatically generate everything right now, but currently we still need some manual work to get a releasable package. Here're the steps of how to get the package.
 
-1. **Add TypeScript emitter dependency in package.json**  
-   
-   In TypeSpec project, modify `package.json` to add dependency for TypeScript emitter, then run `npm install` again to install `@azure-tools/typespec-ts`.
-
-   ```json
-   "dependencies": {
-     ...
-     "@azure-tools/typespec-ts": "latest"
-   },
+1. \*_Configure python emitter in tspconfig.yaml_
+
+   In [Azure/azure-rest-api-specs(https://github.com/Azure/azure-rest-api-specs) repo, there should be tspconfig.yaml where main.tsp of your service is. Make sure it is configured for JavaScript SDK for example:
+
+   ```yaml
+   parameters:
+     "js-sdk-folder":
+       default: "{project-root}/azure-sdk-for-js/"
+     "service-directory-name":
+       default: "YOUR_SERVICE_DIRECRORY"
+   emit:
+     - "@azure-tools/typespec-ts"
+   options:
+     "@azure-tools/typespec-ts":
+       title: SERVICE_NAME
+       emitter-output-dir: "{js-sdk-folder}/sdk/{service-directory-name}/YOUR_PACKAGE_NAME-rest"
+       generateMetadata: true
+       generateTest: true
+       packageDetails:
+         name: "@azure-rest/YOUR_PACKAGE_NAME-rest"
+         description: "Service Description"
+         version: "1.0.0-beta.1"
```

Comment:
We should avoid having the package versions in this configuration data. The version should be controlled in the sdk repo package.json.

Reply:
Currently we need the customer's input to increase/control the package version in config data but I agree we could improve this. Our management-plane experience is to bump the version automatically by the calculated changelog but it is not implemented in DPG side yet.

Regarding `The version should be controlled in the sdk repo package.json.`  do you mean we expect the customer to edit `package.json` to bump the version or expect we should bump the version in our tools?

Reply:
The version should only be in the package.json file and for most data-planes people manually set it or we have automation after release to bump it. For initial project creation we should just set it to 1.0.0-beta.1 and then we can decide if we need folks to manually update it or if we have some tools that will update it for folks but either way it should not be in the tspconfig.yaml file in the specs repo.

Reply:
Agree that we could set the default version as 1.0.0-beta.1 and ask customer to manually set it if they want a different version, which could be done by either set the config in the tspconfig.yaml and rerun the codegen or change it manually. one thing I would like to mention is, we also need the version to generate the user agent prefix inside the client factory https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/purview/purview-workflow-rest/src/purviewWorkflow.ts#L28.

Reply:
For other versions like that we have annotations in the package.json, see https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/purview/purview-workflow-rest/package.json#L109 that get used by our versioning scripts when we set the version it also updates those places. See https://github.com/Azure/azure-sdk-for-js/tree/main/eng/tools/versioning#tracking-format for more information on that.